### PR TITLE
Session resumption

### LINKED ISSUE #196: Provide Guidance on TLS 1.2 resumption
* Overall goal: ensure forward secrecy, possibly also session length
* Revisit sec. 3.4
* https://www.venafi.com/blog/tls-session-resumption

### PR DESCRIPTION
**Pull request by yaronf**
Fixes #196.

### CODE REVIEW

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 1]
```diff
@@ -344,18 +346,32 @@ Implementers should note that compression at higher protocol levels can allow an
...
 * For similar reasons, session ticket validity SHOULD be limited to a reasonable duration (e.g., half as long as ticket key validity).
 
-
-
 Rationale: session resumption is another kind of TLS handshake, and therefore must be as secure as the initial handshake. This document ({{detail}}) recommends the use of cipher suites that provide forward secrecy, i.e. that prevent an attacker who gains momentary access to the TLS endpoint (either client or server) and its secrets from reading either past or future communication. The tickets must be managed so as not to negate this security property.
 
+TLS 1.3 provides the powerful option of forward secrecy even within a long-lived connection
+that is periodically resumed. Section 2.2 of {{RFC8446}} recommends that clients SHOULD
+send an "key_share" when initiating session resumption.
+In order to gain forward secrecy, we recommend that server implementations SHOULD
```
Comment (stpeter) [inline BCP195bis/draft-ietf-uta-rfc7525bis.md:370]: I would delete "we recommend"

**Review by stpeter — APPROVED** [review]

File: BCP195bis/draft-ietf-uta-rfc7525bis.md (hunk 1 above)
Comment (yaronf) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:370]: I used the term specifically here to clarify that this is a new recommendation, in contrast with the client-side recommendation which comes from RFC8446.

Comment (thomas-fossati) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:370]: Maybe: `[...], it is RECOMMENDED that server implementations respond with a "key_share"` ?

Comment (yaronf) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:370]: This document RECOMMENDS that... ?

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 2]
```diff
@@ -344,18 +346,32 @@ Implementers should note that compression at higher protocol levels can allow an
 ## TLS Session Resumption
 {: #rec-resume}
 
-If TLS session resumption is used in the context of TLS 1.2, care ought to be taken to do so safely. In particular, when using session tickets {{?RFC5077}}, the resumption information MUST be authenticated and encrypted to prevent modification or eavesdropping by an attacker. Further recommendations apply to session tickets:
+Session resumption drastically reduces the number of TLS handshakes and thus is an essential
+performance feature for most deployments.
+
+Stateless session resumption with session tickets is a popular strategy. It is specified in
+{{?RFC5077}} for TLS 1.2,
+and the equivalent PSK-based mechanism for TLS 1.3, Section 4.6.1 of {{RFC8446}}.
```
Comment (thomas-fossati) [inline BCP195bis/draft-ietf-uta-rfc7525bis.md:354]:
```suggestion
Stateless session resumption with session tickets is a popular strategy. For TLS 1.2, it is specified in
{{?RFC5077}}.  For TLS 1.3, an equivalent PSK-based mechanism is described in
Section 4.6.1 of {{RFC8446}}.
```

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 3]
```diff
@@ -344,18 +346,32 @@ Implementers should note that compression at higher protocol levels can allow an
...
 
 * For similar reasons, session ticket validity SHOULD be limited to a reasonable duration (e.g., half as long as ticket key validity).
 
-
-
 Rationale: session resumption is another kind of TLS handshake, and therefore must be as secure as the initial handshake. This document ({{detail}}) recommends the use of cipher suites that provide forward secrecy, i.e. that prevent an attacker who gains momentary access to the TLS endpoint (either client or server) and its secrets from reading either past or future communication. The tickets must be managed so as not to negate this security property.
 
+TLS 1.3 provides the powerful option of forward secrecy even within a long-lived connection
+that is periodically resumed. Section 2.2 of {{RFC8446}} recommends that clients SHOULD
+send an "key_share" when initiating session resumption.
```
Comment (thomas-fossati) [inline BCP195bis/draft-ietf-uta-rfc7525bis.md:369]:
```suggestion
send a "key_share" when initiating session resumption.
```

**Review by thomas-fossati — APPROVED** [review]
Looks good content-wise.  I have left a couple of suggestions to improve the flow, although I'm not very sure they necessarily provide an improvement over the status quo.  Feel free to ignore.

**Pull request merged**